lib/bioinfo/deeptools.ml

```diff
@@ -0,0 +1,190 @@
+  | `ratio -> string "ratio"
+  | `subtract -> string "subtract"
+  | `add -> string "add"
+  | `mean -> string "mean"
+  | `reciprocal_ratio -> string "reciprocal_ratio"
+  | `first -> string "first"
+  | `second -> string "second"
+
+let string_rev s =
+  let lgth = String.length s in
+  String.init lgth (fun c -> s.[lgth -1 -c])
+
+let normalizeto1x_expr nbr =
+  let nbr_s = string_rev (string_of_int nbr) in
+  let lgth = String.length nbr_s in
+  let rec compute acc idx nbr_s lgth = match idx with
+    | e when (idx = lgth) -> string (string_rev acc)
+    | l ->
+      if lgth - l <= 3 then
+        let acc = acc ^ String.sub nbr_s l (lgth-l) in
+        compute acc lgth nbr_s lgth
+      else
+        let acc = acc ^ String.sub nbr_s l 3 ^ "," in
+        compute acc (l+3) nbr_s lgth
+  in
+  compute "" 0 nbr_s lgth
+
+
+let bam_gen_cmd ?outfileformat ?scalefactor ?blacklistfilename ?normalizeto1x
+    ?centerreads ?normalizeusingrpkm ?ignorefornormalization ?skipnoncoveredregions
+    ?smoothlength ?extendreads ?ignoreduplicates ?minmappingquality
+    ?samflaginclude ?samflagexclude ?minfragmentlength ?maxfragmentlength
+    cmd_name other_args =
+  cmd cmd_name ~env (
+    List.append [
+      option (opt "--outFileFormat" file_format_expr) outfileformat ;
+      option (opt "--scaleFactor" float) scalefactor ;
+      option (opt "--blackListFileName" string) blacklistfilename ;
+      option (opt "--normalizeTo1x" normalizeto1x_expr) normalizeto1x ;
+      option (flag string "--normalizeUsingRPKM") normalizeusingrpkm ;
+      option (opt "--ignoreForNormalization" (list string ~sep:" ")) ignorefornormalization ;
+      option (flag string "--skipNonCoveredRegions") skipnoncoveredregions ;
+      option (opt "--smoothLength" int) smoothlength ;
+      option (opt "--extendReads" int) extendreads ;
+      option (flag string "--ignoreDuplicates") ignoreduplicates ;
+      option (opt "--minMappingQuality" int) minmappingquality ;
+      option (flag string "--centerReads") centerreads ;
+      option (opt "--samFlagInclude" int) samflaginclude ;
+      option (opt "--samFlagExclude" int) samflagexclude ;
+      option (opt "--minFragmentLength" int) minfragmentlength ;
+      option (opt "--maxfragmentLength" int) maxfragmentlength ;
+    ]
+      other_args
+  )
+
+
+let bamcoverage ?outfileformat ?scalefactor ?filterrnastrand ?binsize ?blacklistfilename
+    ?(threads = 1) ?normalizeto1x ?normalizeusingrpkm ?ignorefornormalization
+    ?skipnoncoveredregions ?smoothlength ?extendreads ?ignoreduplicates
+    ?minmappingquality ?centerreads ?samflaginclude ?samflagexclude
+    ?minfragmentlength ?maxfragmentlength bam =
+  workflow ~descr:"bamcoverage" ~np:threads ~mem:(3 * 1024) [
+    bam_gen_cmd "bamCoverage" [
+      option (opt "--filterRNAstrand" filterRNAstrand_expr) filterrnastrand ;
+      option (opt "--binSize" int) binsize ;
+      opt "--numberOfProcessors" ident np ;
+      string "--bam" ; (dep bam) // "reads.bam"  ;
```

Comment:
rename `bam` into `indexed_bam` so that the actual type is more evident, and use `Samtools.indexed_bam_to_bam ` selector instead of the `//` operator, this is too brittle.